pennylane/optimize/adam.py

```diff
@@ -75,69 +72,46 @@ def apply_grad(self, grad, args):
         args_new = list(args)
 
         if self.accumulation is None:
-            self.accumulation = {"fm": [None] * len(args), "sm": [None] * len(args), "t": 0}
+            self.accumulation = {"fm": [0] * len(args), "sm": [0] * len(args), "t": 0}
 
         self.accumulation["t"] += 1
 
         # Update step size (instead of correcting for bias)
         new_stepsize = (
             self.stepsize
-            * math.sqrt(1 - self.beta2 ** self.accumulation["t"])
+            * sqrt(1 - self.beta2 ** self.accumulation["t"])
             / (1 - self.beta1 ** self.accumulation["t"])
         )
 
         trained_index = 0
         for index, arg in enumerate(args):
-            if getattr(arg, "requires_grad", True):
-                x_flat = _flatten(arg)
-                grad_flat = list(_flatten(grad[trained_index]))
-                trained_index += 1
+            if getattr(arg, "requires_grad", False):
+
+                self._update_accumulation(index, grad[trained_index])
+                args_new[index] = arg - new_stepsize * self.accumulation["fm"][index] / (
+                    sqrt(self.accumulation["sm"][index]) + self.eps
+                )
 
-                self._update_moments(index, grad_flat)
-
-                x_new_flat = [
-                    e - new_stepsize * f / (math.sqrt(s) + self.eps)
-                    for f, s, e in zip(
-                        self.accumulation["fm"][index], self.accumulation["sm"][index], x_flat
-                    )
-                ]
-                args_new[index] = unflatten(x_new_flat, arg)
-
-                if isinstance(arg, ndarray):
-                    # Due to a bug in unflatten, input PennyLane tensors
-                    # are being unwrapped. Here, we cast them back to PennyLane
-                    # tensors.
-                    # TODO: remove when the following is fixed:
-                    # https://github.com/PennyLaneAI/pennylane/issues/966
-                    args_new[index] = args_new[index].view(tensor)
-                    args_new[index].requires_grad = True
+                trained_index += 1
 
         return args_new
 
-    def _update_moments(self, index, grad_flat):
+    def _update_accumulation(self, index, grad):
```

Comment:
Maybe we should rename `accumulation` to `moments` and not the inverse? I think fm and sm stand for that (first order moment and second order moment).

Reply:
This change was to standardize the name of this method across different optimizers.